### PR TITLE
Update serverless docs

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- update serverless docs

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/15002

### CODE REVIEW

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 1]
```diff
@@ -446,20 +440,20 @@ spec:
                 description: Pod selector used to match pods in function deployment
                 type: string
               reference:
-                description: Reference specifies either the branch name, tag or the
-                  commit revision from which the Function Controller automatically
-                  fetches the changes in the Function's code and dependencies.
+                description: Specifies either the branch name, tag or the commit revision
```
Comment (mmitoraj) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:443]:
```suggestion
                description: Specifies either the branch name, tag or commit revision
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 2]
```diff
@@ -245,30 +247,30 @@ type Condition struct {
...
 	Conditions []Condition `json:"conditions,omitempty"`
-	// The repository which was used to build the function
+	// Specify the repository which was used to build the function
 	Repository `json:",inline,omitempty"`
 	// Total number of non-terminated pods targeted by this function
 	Replicas int32 `json:"replicas,omitempty"`
 	// Pod selector used to match pods in function deployment
 	PodSelector string `json:"podSelector,omitempty"`
-	// Commit hash used to build function
+	// Specify the commit hash used to build function
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:271]:
```suggestion
	// Specifies the commit hash used to build the Function.
```

**Review by mmitoraj — COMMENTED** [review]
Additionally, can we also change the descriptions for the following:
- **resourceConfiguration.​build.​resources.​limits**
- **resourceConfiguration.​build.​resources.​requests**
- **resourceConfiguration.​function.​resources.​limits**
- **resourceConfiguration.​function.​resources.​requests**

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 3]
```diff
@@ -178,33 +178,32 @@ spec:
...
-                          of resource. Can't be used at the same time with Resources.
+                        description: Defines the name of the predefined set of values
+                          of resource. Can't be used together with **Resources**.
                         type: string
                       resources:
-                        description: Resources defines amount of resources available
-                          for the Pod to use. Can't be used at the same time with
-                          Profile.
+                        description: Defines amount of resources available for the
+                          Pod to use. Can't be used together with Profile.
```
[the remark this answers is not in the document]
Comment (dbadura) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:206]: It's reference to it has to be capitalized

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 4]
```diff
@@ -149,50 +152,51 @@ const (
...
-	// then the Replicas field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas.
+	// Defines the exact number of Function's Pods to run at a time.
+	// If **ScaleConfig** is configured, or if the Function is targeted by an external scaler,
+	// then the **Replicas** field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas.
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 
 	// +optional
-	// Deprecated: `.spec.Labels` and `.spec.Annotations` should be used to annotate/label function's pods.
+	// Deprecated: **Labels** and **Annotations** should be used to annotate/label function's pods.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:189]:
```suggestion
	// Deprecated: Use **Template.Annotations** and **Template.Labels** to annotate and/or label Function's Pods.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 5]
```diff
@@ -39,33 +39,33 @@ const (
...
+	// Specifies the Function's dependencies.
 	//+optional
 	Dependencies string `json:"dependencies,omitempty"`
 }
 
 type GitRepositorySource struct {
 	// +kubebuilder:validation:Required
 
-	// URL provides the address to the Git repository with the Function's code and dependencies.
+	// URL of the Git repository with the Function's code and dependencies.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:63]:
```suggestion
	// Specifies the URL of the Git repository with the Function's code and dependencies.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 6]
```diff
@@ -232,47 +236,47 @@ const (
...
+	// Specify the repository which was used to build the function.
 	Repository `json:",inline,omitempty"`
-	// Total number of non-terminated pods targeted by this function
+	// Specifies the total number of non-terminated Pods targeted by this Function.
 	Replicas int32 `json:"replicas,omitempty"`
-	// Pod selector used to match pods in function deployment
+	// Specifies the Pod selector used to match Pods in the Function deployment.
 	PodSelector string `json:"podSelector,omitempty"`
-	// Commit hash used to build function
+	// Specifies the commit hash used to build the Function
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:273]:
```suggestion
	// Specifies the commit hash used to build the Function.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 7]
```diff
@@ -232,47 +236,47 @@ const (
 )
 
 type Condition struct {
-	// Type of function condition.
+	// Specifies the type of the Function condition.
 	Type ConditionType `json:"type,omitempty"`
-	// Status of the condition, one of True, False, Unknown.
+	// Specifies the status of the condition. The value is either 'True', 'False', or 'Unknown'.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:241]:
```suggestion
	// Specifies the status of the condition. The value is either `True`, `False`, or `Unknown`.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 8]
```diff
@@ -232,47 +236,47 @@ const (
 )
 
 type Condition struct {
-	// Type of function condition.
+	// Specifies the type of the Function condition.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:239]:
```suggestion
	// Specifies the type of the Function's condition.
```
You were right - it sounds better :)

**Review by mmitoraj — COMMENTED** [review]
Final comments :)

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 9]
```diff
@@ -149,50 +152,51 @@ const (
...
-	// then the Replicas field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas.
+	// Defines the exact number of Function's Pods to run at a time.
+	// If **ScaleConfig** is configured, or if the Function is targeted by an external scaler,
+	// then the **Replicas** field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas.
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 
 	// +optional
-	// Deprecated: `.spec.Labels` and `.spec.Annotations` should be used to annotate/label function's pods.
+	// Deprecated: **Labels** and **Annotations** should be used to and/or label Function's Pods.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:189]:
```suggestion
	// Deprecated: Use **Labels** and **Annotations** to label and/or annotate Function's Pods.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 10]
```diff
@@ -232,47 +236,47 @@ const (
...
+	// Specifies an array of conditions describing the status of the parser.
 	Conditions []Condition `json:"conditions,omitempty"`
-	// The repository which was used to build the function
+	// Specify the repository which was used to build the function.
 	Repository `json:",inline,omitempty"`
-	// Total number of non-terminated pods targeted by this function
+	// Specifies the total number of non-terminated Pods targeted by this Function.
 	Replicas int32 `json:"replicas,omitempty"`
-	// Pod selector used to match pods in function deployment
+	// Specifies the Pod selector used to match Pods in the Function Deployment.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:271]:
```suggestion
	// Specifies the Pod selector used to match Pods in the Function's Deployment.
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 11]
```diff
@@ -232,47 +236,47 @@ const (
...
+	// Specifies the total number of non-terminated Pods targeted by this Function.
 	Replicas int32 `json:"replicas,omitempty"`
-	// Pod selector used to match pods in function deployment
+	// Specifies the Pod selector used to match Pods in the Function's Deployment.
 	PodSelector string `json:"podSelector,omitempty"`
-	// Commit hash used to build function
+	// Specifies the commit hash used to build the Function.
 	Commit string `json:"commit,omitempty"`
-	// Runtime image version used to build and run function pods
+	// Specifies the image version used to build and run the Function Pods.
```
Comment (mmitoraj) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:275]:
```suggestion
	// Specifies the image version used to build and run the Function's Pods.
```

**Review by mmitoraj — APPROVED** [review]

**Review by anoipm — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Pull request merged**